### PR TITLE
Only create JJB update job via puppet module

### PR DESCRIPTION
**Pull request by ehelms**
Now that https://ci.theforeman.org/job/foreman-infra-jenkins-jobs-update/ is handling updates for ci.theforeman.org and ci.centos.org the JJB puppet module need only deploy `foreman-infra-jenkins-jobs-update` itself.

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/manifests/config.pp [hunk 1]
```diff
@@ -22,29 +22,13 @@
     notify  => Exec["jenkins-jobs-update-${config_name}"],
   }
 
-  cron { "jenkins-jobs-update-${config_name}-delete-old":
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/manifests/config.pp:25]: those should be `ensure => absent` to really remove them.

File: puppet/modules/jenkins_job_builder/manifests/config.pp [hunk 2]
```diff
@@ -22,29 +22,13 @@
...
-
   exec { "jenkins-jobs-update-${config_name}":
-    command => "jenkins-jobs --conf ${inifile} update ${directory}/${config_name} > /var/cache/jjb.xml",
+    command => "jenkins-jobs --conf ${inifile} update ${directory}/${config_name} foreman-infra-jenkins-jobs-update",
     timeout => $jenkins_jobs_update_timeout,
     path    => '/bin:/usr/bin:/usr/local/bin',
     require => File[$inifile],
   }
 
-  cron { "remove-unmanaged-jobs-${config_name}":
```
Comment (evgeni) [inline puppet/modules/jenkins_job_builder/manifests/config.pp:40]: those should be `ensure => absent` to really remove them.

**Review by evgeni — CHANGES_REQUESTED** [review]
we can probably also remove the `unmanaged_jobs.rb` file deployment

File: puppet/modules/jenkins_job_builder/manifests/config.pp (hunk 1 above)
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/manifests/config.pp:25]: Ahh right, the ole absent then remove cycle. I forget that some(every)times.

**Review by evgeni — APPROVED** [review]

File: puppet/modules/jenkins_job_builder/manifests/config.pp [hunk 3]
```diff
@@ -23,6 +23,7 @@
   }
 
   cron { "jenkins-jobs-update-${config_name}-delete-old":
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/manifests/config.pp:25]: I won't miss this daily email from cron, but what is now responsible for deleting old (unmanaged) jobs?

File: puppet/modules/jenkins_job_builder/manifests/config.pp [hunk 4]
```diff
@@ -31,13 +32,14 @@
   }
 
   exec { "jenkins-jobs-update-${config_name}":
-    command => "jenkins-jobs --conf ${inifile} update ${directory}/${config_name} > /var/cache/jjb.xml",
+    command => "jenkins-jobs --conf ${inifile} update ${directory}/${config_name} foreman-infra-jenkins-job-update",
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/manifests/config.pp:35]: :-1: please revert. This is a defined type and should be generic. If anything, it should be a parameter.

Comment (ehelms) [reply puppet/modules/jenkins_job_builder/manifests/config.pp:35]: I missed that, gets confusing for me which things are generic and which things are specific for our infra. https://github.com/theforeman/foreman-infra/pull/1556

File: puppet/modules/jenkins_job_builder/manifests/config.pp (hunk 3 above)
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/manifests/config.pp:25]: Is an unmanaged job a job created out of band of JJB?

Comment (evgeni) [reply puppet/modules/jenkins_job_builder/manifests/config.pp:25]: we call `update --delete-old` which removes all jobs that are not currently defined in jjb

Comment (ekohl) [reply puppet/modules/jenkins_job_builder/manifests/config.pp:25]: Can be. It also deleted jobs that were removed from our JJB config.

**Pull request merged**